test-data/run.test

```diff
@@ -2352,6 +2352,67 @@ def f(x: List[Tuple[int, int]]) -> int:
 from native import f
 assert f([(5, 6)]) == 11
 
+[case testUnpack]
+a, *b = [1, 2, 3, 4, 5]
+
+*c, d = [1, 2, 3, 4, 5]
+
+e, *f = [1,2]
+
+j, *k, l = [1, 2, 3]
+
+m, *n, o = [1, 2, 3, 4, 5, 6]
+
+p, q, r, *s, t = [1,2,3,4,5,6,7,8,9,10]
+
+def tests():
```

Comment:
There's no need to do the equality tests here in both the driver and in compiled code, since calculating equality isn't something we are trying to test here.

Whether to do the asserts in the driver code or the test code is a question with some tradeoffs:
 * Asserting in the test code looks better and is easier to read if the asserts are placed closely with the operation. (Doing it that way you'd probably want it all in the `tests` function)
 * Asserting in the driver is in some sense more reliable. One can imagine a codegen bug that results in an assert not tripping even though the calculation was wrong.

We've generally leaned towards doing the asserts in the driver, but the compiler is pretty reliable now and it is pretty unlikely that a codegen bug would prevent these asserts from tripping and not break anything else.

*shrug*